pandas/tests/indexing/test_indexing.py

```diff
@@ -627,6 +627,11 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_empty_index_for_empty_dataframe(self):
+        empty_df = pd.DataFrame(columns=['A'])
+        df_with_empty_value = pd.DataFrame(columns=['A'], data=[])
```

Comment:
Can you include something like the original issue? Something that hits `DataFrame.__setitem__`?

df2 = df.copy()
df['A'] = []
tm.assert_index_equal(df.index, df2.index)
```

I think the other assert can be removed.